crates/sp-domains/src/bundle/bundle_v0.rs

```diff
@@ -0,0 +1,96 @@
+#[cfg(not(feature = "std"))]
+extern crate alloc;
+
+use crate::bundle::OpaqueBundle;
+use crate::execution_receipt::ExecutionReceipt;
+use crate::{HeaderHashFor, HeaderHashingFor, HeaderNumberFor, OperatorSignature, ProofOfElection};
+#[cfg(not(feature = "std"))]
+use alloc::vec::Vec;
+use parity_scale_codec::{Decode, Encode};
+use scale_info::TypeInfo;
+use sp_runtime::OpaqueExtrinsic;
+use sp_runtime::traits::{Hash, Header as HeaderT};
+use sp_weights::Weight;
+
+#[derive(Debug, Decode, Encode, TypeInfo, PartialEq, Eq, Clone)]
+pub struct BundleHeaderV0<Number, Hash, DomainHeader: HeaderT, Balance> {
+    /// Proof of bundle producer election.
+    pub proof_of_election: ProofOfElection,
+    /// Execution receipt that should extend the receipt chain or add confirmations
+    /// to the head receipt.
+    pub receipt: ExecutionReceipt<
+        Number,
+        Hash,
+        HeaderNumberFor<DomainHeader>,
+        HeaderHashFor<DomainHeader>,
+        Balance,
+    >,
+    /// The total (estimated) weight of all extrinsics in the bundle.
+    ///
+    /// Used to prevent overloading the bundle with compute.
+    pub estimated_bundle_weight: Weight,
+    /// The Merkle root of all new extrinsics included in this bundle.
+    pub bundle_extrinsics_root: HeaderHashFor<DomainHeader>,
+}
+
+impl<Number: Encode, Hash: Encode, DomainHeader: HeaderT, Balance: Encode>
+    BundleHeaderV0<Number, Hash, DomainHeader, Balance>
+{
+    /// Returns the hash of this header.
+    pub fn hash(&self) -> HeaderHashFor<DomainHeader> {
+        HeaderHashingFor::<DomainHeader>::hash_of(self)
+    }
+}
+
+/// Header of bundle.
+#[derive(Debug, Decode, Encode, TypeInfo, PartialEq, Eq, Clone)]
+pub struct SealedBundleHeaderV0<Number, Hash, DomainHeader: HeaderT, Balance> {
+    /// Unsealed header.
+    pub header: BundleHeaderV0<Number, Hash, DomainHeader, Balance>,
+    /// Signature of the bundle.
+    pub signature: OperatorSignature,
+}
+
+impl<Number: Encode, Hash: Encode, DomainHeader: HeaderT, Balance: Encode>
+    SealedBundleHeaderV0<Number, Hash, DomainHeader, Balance>
+{
+    /// Constructs a new instance of [`SealedBundleHeaderV0`].
+    pub fn new(
+        header: BundleHeaderV0<Number, Hash, DomainHeader, Balance>,
+        signature: OperatorSignature,
+    ) -> Self {
+        Self { header, signature }
+    }
+}
+
+/// Domain bundle v1.
+#[derive(Debug, Decode, Encode, TypeInfo, PartialEq, Eq, Clone)]
+pub struct BundleV0<Extrinsic, Number, Hash, DomainHeader: HeaderT, Balance> {
```

Comment:
<img width="474" height="267" alt="image" src="https://github.com/user-attachments/assets/0af7f065-2c32-43bb-a2a8-e9ec091caa37" />

The mismatch between the version number in the comment and the type name needs fixing (or more explanation).